src/pages/api/defineTimeRestrictions.ts

```diff
@@ -183,35 +26,47 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
             throw new Error('Failed to retrieve the fareType attribute for the defineTimeRestrictions API');
         }
 
-        const filteredReqBody = formatRequestBody(req);
-
-        const { timeRestriction, startTime, endTime, validDaysSelected, validDays } = filteredReqBody;
-
-        const initialErrors = await runValidationSchema(defineTimeRestrictionsSchema, filteredReqBody);
-        const combinedErrors = await runValidationSchema(
-            timeRestrictionConditionalInputSchema,
-            filteredReqBody,
-            initialErrors,
-        );
+        const { validDaysSelected, validDays } = req.body;
 
         const timeRestrictionsDefinition: TimeRestriction = {
-            startTime,
-            endTime,
-            validDays,
+            validDays: [],
         };
 
-        if (combinedErrors.length > 0) {
+        if (!validDaysSelected) {
             const timeRestrictionsDefinitionWithErrors: TimeRestrictionsDefinitionWithErrors = {
-                ...timeRestrictionsDefinition,
-                timeRestriction,
+                validDays,
                 validDaysSelected,
-                errors: combinedErrors,
+                errors: [{ errorMessage: 'Choose one of the options below', id: 'valid-days-required' }],
             };
             updateSessionAttribute(req, TIME_RESTRICTIONS_DEFINITION_ATTRIBUTE, timeRestrictionsDefinitionWithErrors);
             redirectTo(res, '/defineTimeRestrictions');
             return;
         }
 
+        if (validDaysSelected === 'Yes') {
```

Comment:
In the case where this is 'No', we should clear down the FULL_TIME_RESTRICTIONS attribute in case a user goes back to change it